tools/mmp/driver.cs

```diff
@@ -1483,13 +1491,22 @@ static void CopyI18nAssemblies (I18nAssemblies i18n)
 				resolved_assemblies.Add (Path.Combine (fx_dir, "I18N.West.dll"));
 		}
 
+		static void CopyFileAndRemoveReadOnly (string src, string dest) {
+			File.Copy (src, dest, true);
+
+			FileAttributes attrs = File.GetAttributes (dest);
+			if ((attrs & FileAttributes.ReadOnly) == FileAttributes.ReadOnly)
+				File.SetAttributes (dest, attrs & ~FileAttributes.ReadOnly);
+		}
+
 		static void CopyAssemblies () {
 			foreach (string asm in resolved_assemblies) {
 				var mdbfile = string.Format ("{0}.mdb", asm);
 				var configfile = string.Format ("{0}.config", asm);
 				string filename = Path.GetFileName (asm);
 
-				File.Copy (asm, Path.Combine (mmp_dir, filename), true);
+				// The linker later gets angry if you copy in a read only assembly
+				CopyFileAndRemoveReadOnly (asm, Path.Combine (mmp_dir, filename));
```

Comment:
I'm not sure it's the best place to solve this as it leave the 3 other linkers (XI, XA and monolinker) code base with an identical issue.